src/v/kafka/server/handlers/produce.cc

```diff
@@ -632,6 +632,12 @@ produce_handler::handle(request_context ctx, ss::smp_service_group ssg) {
     produce_request request;
     request.decode(ctx.reader(), ctx.header().version);
     log_request(ctx.header(), request);
+
+    if (ctx.recovery_mode_enabled()) {
```

Comment:
nit: since this is a performance critical path we may use `unlikely`

Reply:
fixed